tests/test_enterprise/api/test_views.py

```diff
@@ -3469,6 +3473,38 @@ def test_bulk_enrollment_in_bulk_courses_pending_licenses(
         else:
             mock_track_enroll.assert_not_called()
 
+        # verify notification sent correctly for each course to applicable learners
+        if 'notify' in body:
```

Comment:
nit: This is already a long test and I don't love conditional logic in tests, it makes them breakable. Can you please refactor the existing test by splitting it in half (old_test = 1 new function, if conditional code into 2nd new function) and then make it like this: 

```

def _original_test_logic():
   stuff

def notify_test():
   your new code

@<data>
def test_bulk_enrollment_in_bulk_courses_pending_licenses():
  original_code(params)
  
@<data you want to use for notifications>
def test_bulk_enrollment_in_bulk_courses_pending_licenses_with_notifications():
  original_code(params)
  new_notification_code(params)

```


Another option is to just run the notify verification on every test case.

Reply:
I like this! let me get back with this change, so the notification use case testing is separate

Reply:
done!